docs/book/src/developer/providers/v1alpha3-to-v1alpha4.md

```diff
@@ -0,0 +1,35 @@
+# Cluster API v1alpha3 compared to v1alpha4
+
+## Minimum Go version
+
+- The Go version used by Cluster API is now Go 1.15+
+
+## Controller Runtime version
+
+- The Controller Runtime version is now v0.7.+
+
+## Kind version
+
+- The KIND version used for this release is v0.9.x
+
+##
+
+## Upgrade kube-rbac-proxy to v0.5.0
+
+- Find and replace the `kube-rbac-proxy` version (usually the image is `gcr.io/kubebuilder/kube-rbac-proxy`) and update it to `v0.5.0`.
+
+## The controllers.DeleteNodeAnnotation constant has been removed
+
+- This annotation `cluster.k8s.io/delete-machine` was originally deprecated a while ago when we moved our types under the `x-k8s.io` domain.
```

Comment:
Not directly related to provider developers, but we should track having a release note for v1alpha4 telling users that external tooling using this annotation should ensure they are using the correct annotation.

For example, if using cluster-autoscaler, users will need to ensure they are using a version that uses the right annotations, such as v1.20.0, v1.19.1, v1.18.3, v1.17.4 compared to earlier versions.

Is there somewhere that we are tracking user-facing release notes?

Reply:
We have the PR marked as ⚠️, it'll show up once we generate the releae notes